### PR TITLE
WIP: enhance parser legacy support and adding more comment

### PR DESCRIPTION
**Pull request by Shikanime**
Impl #101: Best effort for legacy text parsing
Impl #99 + #84: To be done
Concern #102

### CODE REVIEW

File: src/app/editor/viewer/lib/parsers/sketch-style-parser.service.ts [hunk 1]
```diff
@@ -18,74 +19,91 @@ export enum BorderType {
...
-        if (typeof obj[property] === 'object') {
-          // visit child
-          if (obj[property].frame && obj[property].frame._class === 'rect') {
-            this.visitObject(obj[property], obj, obj[property]);
+  /**
+   * Recurse over all layers properties and parse
+   * valuable class to CSS
+   */
+  visitObject(curr: any, parent: any, root: SketchMSLayer) {
+    for (const property in curr) {
```
Comment (manekinekko) [inline src/app/editor/viewer/lib/parsers/sketch-style-parser.service.ts]: what's curr ?

File: src/app/editor/viewer/lib/parsers/sketch-style-parser.service.ts [hunk 2]
```diff
@@ -94,131 +112,91 @@ export class SketchStyleParserService {
...
-      const parsedArchivedAttributedString = this.binaryPlistParser.parse64Content(
-        archivedAttributedString
-      );
-      (root.attributedString
-        .archivedAttributedString as any)._transformed = parsedArchivedAttributedString;
+  parseAttributeString(curr: SketchMSLayer, root: SketchMSLayer) {
+    const obj = curr.attributedString;
+    if (obj.hasOwnProperty('archivedAttributedString')) {
+      const archive = this.binaryPlistParser.parse64Content(obj.archivedAttributedString._archive);
+      switch (archive.$key) {
```
Comment (manekinekko) [inline src/app/editor/viewer/lib/parsers/sketch-style-parser.service.ts]: we should check that `archive` is truthy.

File: src/app/editor/viewer/lib/parsers/sketch-style-parser.service.ts (hunk 1 above)
Comment (Shikanime) [reply src/app/editor/viewer/lib/parsers/sketch-style-parser.service.ts]: Oh, it was supposed to be obj, I'll change it.

File: src/app/editor/viewer/lib/parsers/sketch-style-parser.service.ts (hunk 2 above)
Comment (Shikanime) [reply src/app/editor/viewer/lib/parsers/sketch-style-parser.service.ts]: It should be safer, i will do it.

File: src/app/editor/viewer/lib/parsers/sketch-style-parser.service.ts (hunk 1 above)
Comment (manekinekko) [reply src/app/editor/viewer/lib/parsers/sketch-style-parser.service.ts]: then just name it `currentObject` :)

**Comment by Shikanime** [comment]
@manekinekko As you can see [here](https://github.com/sfeir-open-source/xlayers/blob/c97e5472a1d66994541ea9f8f0249a4e77a58503/src/app/editor/viewer/lib/sketch-layer.component.ts#L114), I want to remove everything that is based on the AST SketchApp.
I want us to produce our own CSS-based AST instead of enriching the AST, such as:

```json
{
  "pages": [
    {
      "scopes": [
        {
          "content": "some text",
          "style": {
            "x-rule": "x-value"
          }
        }
      ]
    }
  ]
}
```

scopes: SketchMSLayer or SketchMSRect

What do you think of design? This will be a breaking change and should therefore be scheduled for the next major version after the implementation of Vue/React/Angular.

**Comment by Shikanime** [comment]
Because of difficulties i decided to switch back to current parser  #157

**Pull request closed without merging**